bql/planner/data_access_test.go

```diff
@@ -463,12 +485,49 @@ func TestDataAccessTripleToRowPredicateBindings(t *testing.T) {
 	}
 }
 
+func TestDataAccessTripleToRowPredicateBindingsError(t *testing.T) {
+	n, pImmutable, _ := testNodePredicateLiteral(t)
+
+	testTable := []struct {
+		t   string
+		cls *semantic.GraphClause
+	}{
+		{
+			t: fmt.Sprintf("%s\t%s\t%s", n, pImmutable, n),
+			cls: &semantic.GraphClause{
+				PAnchorBinding: "?anchorBinding",
+			},
+		},
+		{
+			t: fmt.Sprintf("%s\t%s\t%s", n, pImmutable, n),
+			cls: &semantic.GraphClause{
+				PAnchorAlias: "?anchorAlias",
+			},
+		},
+	}
+
+	for _, entry := range testTable {
+		// Setup for test:
+		tpl, err := triple.Parse(entry.t, literal.DefaultBuilder())
+		if err != nil {
+			t.Fatalf(`triple.Parse failed for triple "%s": %v`, entry.t, err)
+		}
+
+		// Actual test:
+		_, err = tripleToRow(tpl, entry.cls)
+		if _, ok := err.(*skippableError); !ok {
+			t.Errorf(`tripleToRow("%s", %q) = _, %v; want _, skippableError`, tpl, entry.cls, err)
+		}
+	}
+}
```

Comment:
For subtests, go has a Run method that allow us to distinguish between the subtests when the test fails.

Here's an example:

func TestGroupedParallel(t *testing.T) {
    for _, tc := range testCases {
        tc := tc // capture range variable
        t.Run(tc.Name, func(t *testing.T) {
            if got := foo(tc.in); got != tc.out {
                t.Errorf("got %v; want %v", got, tc.out)
            }
            ...
        })
    }
}

More info here: https://golang.org/pkg/testing/#hdr-Subtests_and_Sub_benchmarks

This allows you to define a name for each test case and t.Run will print it separately for you when one of the subtests fails.

Reply:
Thank you for the hint Jose, It is cool that Go has this, I did not know about that.
For the example above with the separation between "setup" with `t.Fatalf` and "actual test" with `t.Errof` along with the given error messages I believe it is clear what each test is about while also providing the tools for the user to understand where exactly was the error (in the case it happened), but it would be nice indeed to have a name for each test case as well to complement that (using `t.Run`).
This current version above is following the standard for test functions in this file and the other files in BadWolf, so I will let it like this for now but I will keep the `t.Run` in mind for a future overall refactor (to include names for each test case, for this and the other test functions as well).

Reply:
Issue created to track this: https://github.com/google/badwolf/issues/152.